pulsar-metadata/src/main/java/org/apache/pulsar/metadata/api/MetadataStore.java

```diff
@@ -159,4 +159,10 @@ public interface MetadataStore extends AutoCloseable {
      * @return the metadata cache object
      */
     <T> MetadataCache<T> getMetadataCache(MetadataSerde<T> serde);
+
+    /**
+     * Async close the current instance.
+     * @return
+     */
+    CompletableFuture<Void> closeAsync();
```

Comment:
Changing the MetadataStore interface would require a PIP. We cannot do that. Please revert this part of the changes. It will also resolve the unclosed metadata store threads issue seen in the thread dumps.

Reply:
I will revert it but it is unrelated with the thread leak.

`metadata-store-coordination-service` threads come from `CoordinationServiceImpl` and the executor is never shutdown.